solarforecastarbiter/io/fetch/nwp.py

```
@@ -0,0 +1,669 @@
+    if len(list(
+            (modelpath / inittime.strftime('%Y/%m/%d/%H')).glob('*.nc'))) > 0:
+        return await next_run_time(inittime, modelpath, model)
+    await sleep_until_inittime(inittime, model)
+    return inittime
+
+
+async def run(basepath, model_name, chunksize, once=False):
+    session = make_session()
+    modelpath = basepath / model_name
+    model = model_map[model_name]
+    inittime = await startup_find_next_runtime(modelpath, session, model)
+    while True:
+        fetch_tasks = set()
+        async for params in files_to_retrieve(session, model, inittime):
+            fetch_tasks.add(asyncio.create_task(
+                fetch_grib_files(session, params, modelpath, inittime,
+                                 chunksize)))
+        files = await asyncio.gather(*fetch_tasks)
+        if len(files) != 0:  # skip to next inittime
+            path_to_files = files[0].parent
+            nctmpfile = await process_grib_to_netcdf(path_to_files)
+            try:
+                await optimize_netcdf(
+                    nctmpfile, path_to_files / f'{model_name}.nc')
+            except Exception:
+                raise
+            else:
+                # remove grib files
+                for f in files:
+                    f.unlink()
+        if once:
+            break
+        else:
+            logger.info('Moving on to next model run')
+            inittime = await next_run_time(inittime, modelpath, model)
+    await session.close()
+
+
+async def optimize_only(path_to_files, model_name):
+    nctmpfile = await process_grib_to_netcdf(path_to_files)
+    try:
+        await optimize_netcdf(
+            nctmpfile, path_to_files / f'{model_name}.nc')
+    except Exception:
+        raise
+    else:
+        # remove grib files
+        for f in path_to_files.glob('*.grib2'):
+            f.unlink()
+
+
+def check_wgrib2():
+    if shutil.which('wgrib2') is None:
+        logger.error('wgrib2 was not found in PATH and is required')
```

is this pulled in through the python dependencies? If not, `pip install .[fetch]` should fail if it's not available.

---

Though we can leave that for a future issue, too.

---

future #64